### ISSUE TITLE
Monster Truck Madness 64 - overlapping text tile?

### ISSUE THREAD

**Issue by olivieryuyu** [gonetz/GLideN64#25]
The text on the game menu seems to be wrapped or overlapped by another one

Angrylion: I solved the problem, in a very logical way. The thing is, if texel1 is used in the 2nd cycle of the combiner, it is in fact texel0 (what Ziggy called 'texel swapping'), BUT for the next pixel (s + ds, t + dt), not for the current pixel.

**Comment by oddMLan** [comment]
![gliden64_mtm64_000](https://cloud.githubusercontent.com/assets/7278372/15261962/4e0fc518-1914-11e6-9ede-77b80a8c9c2a.png)
This is still actual

**Comment by gonetz** [comment]
I know. Angrylion's solution need to be implemented.

**Comment by standard-two-simplex** [comment]
> Angrylion: I solved the problem, in a very logical way. The thing is, if texel1 is used in the 2nd cycle of the combiner, it is in fact texel0 (what Ziggy called 'texel swapping'), BUT for the next pixel (s + ds, t + dt), not for the current pixel.

Can I get any confirmation that this is how it should work?
- If it is cycle 0 
    - If combiner uses Texel 0, use tile described by PrimTile.
    - If combiner uses Texel 1, use tile described by PrimTile + 1.
- If it is cycle 1
    - If combiner uses Texel 0, use tile described by PrimTile +1.
    - If combiner uses Texel 1, use tile described by PrimTile, but texture coordinates of the next pixel.

@gonetz How does the texel swapping happen currently?

**Comment by standard-two-simplex** [comment]
I wrote some proof of concept code on how this can be implemented. However, I believe that combiner code could be modified so that this idea fits in more naturally, but I'm not sure how.

**Comment by gonetz** [comment]
> Can I get any confirmation that this is how it should work?

Correct.

> How does the texel swapping happen currently?

- If it is cycle 1
  - If combiner uses Texel 0, use tile described by PrimTile +1.
  - If combiner uses Texel 1, use tile described by PrimTile, without texture coordinates modification

**Comment by gonetz** [comment]
> I wrote some proof of concept code on how this can be implemented. However, I believe that combiner code could be modified so that this idea fits in more naturally, but I'm not sure how.

Interesting approach. It works, but only in native resolution. And the title screen with copyrights text has issues with texture mapping even in native res.
I also took several attempts in the past to implement that bias to texel one, but result was the same: it either works for title screen or for in-game text and only in native res.

**Comment by standard-two-simplex** [comment]
> Interesting approach. It works, but only in native resolution. And the title screen with copyrights text has issues with texture mapping even in native res.

You're right, non native resolutions still have artifacts and even native resolution has artifacts in certain places.

I added a hack to the use_screen_coords branch to check the behaviour there and got correct picture in both menus and in-game text (but only in native). So I'm guessing there are actually two issues with this game: a general bias also present in other games (like the one in Ogre Battle) and the texel swapping bias described in this issue.
